qiskit/assembler/assemble_schedules.py

```diff
@@ -232,18 +232,25 @@ def _validate_meas_map(instruction_map: Dict[Tuple[int, instructions.Acquire],
         QiskitError: If the instructions do not satisfy the measurement map.
     """
     meas_map_sets = [set(m) for m in meas_map]
-
-    # Check each acquisition time individually
-    for _, instrs in instruction_map.items():
```

Comment:
There may be a more efficient way to do this. You might want to think about starting from scratch to get a clearer implementation -- the old one might be getting in your way.

Check out what `instruction_map` is giving you. Previously, the times were ignored. In `for _, instrs`, the `_` is a tuple which should contain the start time of the instruction, if the docstring is correct! So the work being done to create `time_mapping` has been done previously by the caller.

See what you can do by thinking through that -- try to come up with the lowest time complexity you think is possible. Think about what minimum information you need to decide if the schedule is invalid.

We can chat on webex, if you like, whenever you're ready.